controllers/templatesync/template_sync.go

```diff
@@ -866,3 +1206,116 @@ func getLatestStatusMessage(pol *policiesv1.Policy, tIndex int) string {
 
 	return tmplDetails.History[0].Message
 }
+
+// hasPolicyTemplateLabel queries the CRD for a given GroupVersionResource and returns whether it
+// has the policy-type=template label and an error if the CRD could not be retrieved.
+func (r *PolicyReconciler) hasPolicyTemplateLabel(
+	ctx context.Context, rsrc schema.GroupVersionResource,
+) (bool, error) {
+	crd := extensionsv1.CustomResourceDefinition{}
+	crdName := types.NamespacedName{
+		Name: rsrc.GroupResource().String(),
+	}
+
+	err := r.Get(ctx, crdName, &crd)
+	if err != nil {
+		// If it wasn't found, then it wasn't in the cache and doesn't have the label
+		if k8serrors.IsNotFound(err) {
+			return false, nil
+		}
+
+		return false, err
+	}
+
+	crdLabels := crd.GetLabels()
+	if crdLabels == nil {
+		crdLabels = map[string]string{}
```

Comment:
You should be able to read a nil map just fine so I would skip this.

Reply:
Ah, I didn't know an empty map could tolerate being read.

So I guess this `if` can be removed also, then:
```go
// Handle adding metadata labels
labels := tObjectUnstructured.GetLabels()
if labels == nil {
  labels = map[string]string{}
}
```
https://github.com/dhaiducek/governance-policy-framework-addon/blob/7062a5ea54db942ad555dc7d8562f07d8ae5a8ea/controllers/templatesync/template_sync.go#L501-L503

Reply:
@dhaiducek you'll need to keep that one because you can't write to a nil map.